src/portal/lib/src/image-name-input/image-name-input.component.ts

```diff
@@ -0,0 +1,103 @@
+import { Component, OnDestroy, OnInit } from "@angular/core";
+import { Project } from "../project-policy-config/project";
+import { Observable, Subject } from "rxjs/index";
+import { debounceTime, distinctUntilChanged } from "rxjs/operators";
+import { ProjectService } from "../service/project.service";
+import { AbstractControl, FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { ErrorHandler } from "../error-handler/error-handler";
+
+@Component({
+    selector: "hbr-image-name-input",
+    templateUrl: "./image-name-input.component.html",
+    styleUrls: ["./image-name-input.component.scss"]
+})
+export class ImageNameInputComponent implements OnInit, OnDestroy {
+    noProjectInfo = "";
+    selectedProjectList: Project[] = [];
+    proNameChecker: Subject<string> = new Subject<string>();
+    imageNameForm: FormGroup;
+
+    constructor(
+        private fb: FormBuilder,
+        private errorHandler: ErrorHandler,
+        private proService: ProjectService,
+    ) {
+        this.imageNameForm = this.fb.group({
+            projectName: ["", Validators.required],
+            repoName: ["", Validators.required],
+            tagName: ["", Validators.required],
+        });
+    }
+    ngOnInit(): void {
+        this.proNameChecker
+            .pipe(debounceTime(500))
+            .pipe(distinctUntilChanged())
+            .subscribe((name: string) => {
+                this.noProjectInfo = "";
+                this.selectedProjectList = [];
+                const prolist: any = this.proService.listProjects(name, undefined);
+                if (prolist.subscribe) {
+                    prolist.subscribe(response => {
+                        if (response) {
+                            this.selectedProjectList = response.slice(0, 10);
```

Comment:
this.selectedProjectList = response.slice(9);
Should use 9 to keep 10 values. 
Please add comments and make 9 as a const.

Reply:
It seems `response.slice(0, 10)` returns 10 elements as expected.